src/ripple/basics/base_uint.h

```diff
@@ -510,7 +510,7 @@ inline const base_uint<Bits, Tag> operator+ (
 template <std::size_t Bits, class Tag>
 inline std::string to_string (base_uint<Bits, Tag> const& a)
 {
-    return strHex (a.begin (), a.size ());
+    return strHex (a.cbegin (), a.cend ());
```

Comment:
It's my understanding that the `c` variants of `begin` and `end` have falled out of favor. Is there a particular reason to use them here? Certainly it's clear to the compiler that _ONLY_ the `const` variants can be invoked since the object they're invoked against is `const`.

Reply:
I'd be interested to hear the indicators that `cbegin()` and `cend()` are falling out of favor.  Not disagreeing, but I haven't personally seen that.

It's possible that in C++17 they will be less useful because of the availability of `as_const()` (https://en.cppreference.com/w/cpp/utility/as_const).  But that's no available to us yet.  Is that what you're thinking of, or is there something else?  Thanks.

On this particular topic, I kinda like using `cbegin()` and `cend()` here.  They are really not necessary right now since, as you note, `a` is `const&`.  But suppose someone in the future switches to passing `a` by value.  If that were to happen then `cbegin()` and  `cend()` would still protect `a` from being modified.  To me it just feels like a little bit of non-critical future-proofing.  YMMV.

Reply:
Thanks.  I can easily imagine someone saying that since `for (auto x : y)` doesn't support `cbegin()` and `cend()` that `cbegin` and `cend` are now irrelevant.  I wouldn't, personally, agree with that assessment.  But I can certainly imagine someone promoting that.  I appreciate the background info.

Reply:
That does make sense. I'm sold ;)